uwrt_mars_rover_autonomy/spiralSearch/src/spiralSearch_server.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+Action server for the spiral search algorithm
+Currently works with turtlesim for testing 
+*/
+
+#include "ros/ros.h"
+#include <actionlib/server/simple_action_server.h>
+#include <spiralSearch/spiralSearchAction.h>
+#include "geometry_msgs/Twist.h" // contains the ros message type Twist
+#include "turtlesim/Pose.h" // this allows the node read the pose message type published by the node
+#include <sstream> 
+#include <cmath>
+
+//Action Server variables 
+typedef actionlib::SimpleActionServer<spiralSearch::spiralSearchAction> Server;
+spiralSearch::spiralSearchFeedback feedback; 
+spiralSearch::spiralSearchResult result; 
+
+const double PI = 3.14159265359;
+
+//To publish commands to turtlesim
+ros::Publisher velocity_publisher;
```

Comment:
ideally we don't want global variables like this, you should try to make the action server a class and have the publisher and subscriber as member variables. It makes it more modular when adding to it in the future